templates/flutter/test/services/service_test.dart.twig

```diff
@@ -0,0 +1,107 @@
+    String? paramName,
+    String? idParamName,
+    Map<String, String>? headers,
+    Function(UploadProgress)? onProgress,
+  }) async {
+    return super.noSuchMethod(Invocation.method(#chunkedUpload, [path, params, paramName, idParamName, headers]), returnValue: Response(data: {}));
+  }
+}
+
+void main() {
+    group('{{service.name | caseUcfirst}} test', () {
+        late MockClient client;
+        late {{service.name | caseUcfirst}} {{service.name | caseCamel}};
+
+        setUp(() {
+            client = MockClient();
+            {{service.name | caseCamel}} = {{service.name | caseUcfirst}}(client);
+        });
+
+{% for method in service.methods %}
+        test('test method {{method.name | caseCamel}}()', () async {
+            {%- if method.type == 'webAuth' -%}
+            {%~ elseif method.type == 'location' -%}
+            final Uint8List data = Uint8List.fromList([]);
+            {%- else -%}
+
+                {%~ if method.responseModel and method.responseModel != 'any' ~%}
+            final Map<String, dynamic> data = {
+                {%- for definition in spec.definitions ~%}{%~ if definition.name == method.responseModel -%}{%~ for property in definition.properties | filter((param) => param.required) ~%}
+                '{{property.name | escapeKeyword | escapeDollarSign}}': {% if property.type == 'object' %}<String, dynamic>{}{% elseif property.type == 'array' %}[]{% elseif property.type == 'string' %}'{{property.example}}'{% elseif property.type == 'boolean' %}true{% else %}{{property.example}}{% endif %},{%~ endfor ~%}{% set break = true %}{%- else -%}{% set continue = true %}{%- endif -%}{%~ endfor -%}
+
+            };
+                {%~ else ~%}
+            final data = '';
+                {%- endif -%}
+            {% endif %}
+
+            {%~ if method.type == 'webAuth' ~%}
+            when(client.webAuth(
+                Uri(),
+            )).thenAnswer((_) async => 'done');
+            {%~ elseif 'multipart/form-data' in method.consumes ~%}
+            when(client.chunkedUpload(
+                path: argThat(isNotNull),
+                params: argThat(isNotNull),
+                paramName: argThat(isNotNull),
+                idParamName: argThat(isNotNull),
+                headers: argThat(isNotNull),
+            )).thenAnswer((_) async => Response(data: data));
+            {%~ else ~%}
+            when(client.call(
+                HttpMethod.{{method.method | caseLower}},
+            )).thenAnswer((_) async => Response(data: data));
+            {%~ endif ~%}
+
+            final response = await {{service.name | caseCamel}}.{{method.name | caseCamel}}({%~ for parameter in method.parameters.all | filter((param) => param.required) ~%}
+                {{parameter.name | escapeKeyword | caseCamel}}: {% if parameter.type == 'object' %}{}{% elseif parameter.type == 'file' %}InputFile.fromPath(path: './image.png'){% else %}'{{parameter.example}}'{%~ endif ~%},{%~ endfor ~%}
```

Comment:
This needs to include a `{% elseif parameter.type == 'array' %}[]` or else the teams_test.dart will fail:

<img width="684" alt="image" src="https://github.com/appwrite/sdk-generator/assets/1477010/4e3b3140-2843-4597-8339-b9e80d5670d3">

```suggestion
                {{parameter.name | escapeKeyword | caseCamel}}: {% if parameter.type == 'object' %}{}{% elseif parameter.type == 'array' %}[]{% elseif parameter.type == 'file' %}InputFile.fromPath(path: './image.png'){% else %}'{{parameter.example}}'{%~ endif ~%},{%~ endfor ~%}
```